Source/SA/DPD/DPD.cs

```diff
@@ -0,0 +1,207 @@
+            MemoryPolynomialLeadMemoryDepth, MemoryPolynomialLagMemoryDepth, MemoryPolynomialMaxLead, MemoryPolynomialMaxLag;
+            public static MemoryPolynomialConfiguration GetDefault()
+            {
+                return new MemoryPolynomialConfiguration
+                {
+                    MemoryModelCorrectionType = RFmxSpecAnMXDpdApplyDpdMemoryModelCorrectionType.MagnitudeAndPhase,
+                    IterativeDpdEnabled = RFmxSpecAnMXDpdIterativeDpdEnabled.False,
+                    NumberOfIteration = 3,
+                    MemoryPolynomialDepth = 2,
+                    MemoryPolynomialOrder = 3,
+                    MemoryPolynomialLeadOrder = 2,
+                    MemoryPolynomialLagOrder = 2,
+                    MemoryPolynomialLeadMemoryDepth = 2,
+                    MemoryPolynomialLagMemoryDepth = 2,
+                    MemoryPolynomialMaxLead = 2,
+                    MemoryPolynomialMaxLag = 2
+                };
+            }
+        }
+
+        public struct LookupTableResults
+        {
+            public ComplexWaveform<ComplexSingle> WaveformWithDpdComplexSingle;
+            public float[] LookupTableInputPowers;
+            public ComplexSingle[] LookupTableComplexGains;
+            public double PostDPDPAPR;
+            public double PowerOffset;
+        }
+
+        public struct MemoryPolynomialResults
+        {
+            public ComplexWaveform<ComplexSingle> WaveformWithDpdComplexSingle;
+            public ComplexSingle[] DpdPolynomial;
+            public double PostDPDPAPR;
+            public double PowerOffset;
+        }
+        #endregion
+        #region ConfigureDPD
+        public static void ConfigureCommon(RFmxSpecAnMX specAnSignal, CommonConfiguration commonConfig, Waveform waveform, string selectorString = "")
+        {
+            RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent idlePresent;
+     
+            idlePresent = waveform.IdleDurationPresent ? RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent.True : RFmxSpecAnMXDpdReferenceWaveformIdleDurationPresent.False;
+
+            specAnSignal.SelectMeasurements(selectorString, RFmxSpecAnMXMeasurementTypes.Dpd, true);
+
+            specAnSignal.Dpd.Configuration.ConfigureReferenceWaveform(selectorString, waveform.WaveformData, idlePresent, commonConfig.SignalType);
+            specAnSignal.Dpd.Configuration.ConfigureDutAverageInputPower(selectorString, commonConfig.DutAverageInputPower);
+            specAnSignal.Dpd.Configuration.ConfigureMeasurementInterval(selectorString, commonConfig.DpdMeasurementInterval);
+            specAnSignal.Dpd.Configuration.ConfigureMeasurementSampleRate(selectorString, commonConfig.DpdSampleRateMode, commonConfig.SampleRate);
+
+        }
+
+        public static void ConfigureRFmxLUT(RFmxSpecAnMX specAnSignal, LookupTableConfiguration lutConfig, string selectorString = "")
```

Comment:
I think that this should be changed to the below. @smooresni agree?
```suggestion
        public static void ConfigureLookupTable(RFmxSpecAnMX specAnSignal, LookupTableConfiguration lutConfig, string selectorString = "")
```

Reply:
@mdwhitten agreed

Reply:
@cindyxie why did you add "DPD" to the end? Suggest removing that.